tracer/src/Datadog.Trace/DuckTyping/DuckType.cs

```diff
@@ -1013,6 +1026,23 @@ public static T Create(object instance)
                 return GetProxy(instance.GetType()).CreateInstance<T>(instance);
             }
 
+            /// <summary>
+            /// Create a new instance of a proxy type for a target instance using the T proxy definition
+            /// </summary>
+            /// <typeparam name="TOriginal">The original instance's type </typeparam>
+            /// <param name="instance">Object instance</param>
+            /// <returns>Proxy instance</returns>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static T CreateFrom<TOriginal>(TOriginal instance)
+            {
+                if (instance is null)
+                {
+                    return default;
+                }
+
+                return GetProxy(typeof(TOriginal)).CreateInstance<T, TOriginal>(instance);
+            }
```

Comment:
I think we can simplify this by:
- Adding an struct constraint over TOriginal (Is the only use we do).
- Remove the conditional block.

Reply:
Like this?

```csharp
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public static T CreateFrom<TOriginal>(TOriginal instance)
    where TOriginal : struct
{
    return GetProxy(typeof(TOriginal)).CreateInstance<T, TOriginal>(instance);
}
```

If this is for `struct` only, do you think we should improve the name? `CreateFromValueType` perhaps?

Reply:
Yes to both, that's the implementation and yeah I agree we should go for a better name, `CreateFromValueType` sounds good.

Reply:
So, apparently, you can't use `where T : struct` with nullable types 🤷‍♂️ 

![image](https://user-images.githubusercontent.com/18755388/154241919-ed766999-571d-45fc-a7a6-cdc5aa8598c1.png)

Which means `CreateFromValueType` isn't a good name either 😄

Reply:
@kevingosse pointed out that we _can_ have a nullable overload, so maybe we should rethink this again later?

```csharp
public Main()
{
    int? a = 0;

    Test(a);
}

private void Test<T>(T? a) where T : struct
{
}

private void Test<T>(T a) where T : struct
{
}
```